### DISCUSSION TITLE
Can mkdocstrings generate python strings for an entire codebase?

### DISCUSSION THREAD

**Discussion by bendnorman** [mkdocstrings/mkdocstrings#715] [Q&A] [answered]
Is it possible for `mkdocstrings` to traverse an entire python code base and generate docs for all of the modules? I've managed to manually reference doc strings in markdown files and have `mkdocstrings` generate docs.

**Comment by pawamoy** [comment, marked as answer]
Hi @bendnorman, yes, it's possible! Please see https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages.

A more integrated and robust solution was requested here: https://github.com/mkdocstrings/mkdocstrings/issues/389. In this issue we mention https://github.com/jcayers20/mkdocs-autoapi, which could be of interest to you :slightly_smiling_face:

**Reply by bendnorman** [reply]
Thanks for the quick response @pawamoy! This solution worked for me. I'll take a look at the `mkdocs-autoapi` repo as well.

**Comment by KINGOFKINGOFKEY** [comment]
Install MkDocs and mkdocstrings:
sh
pip install mkdocs mkdocstrings
Set Up Your MkDocs Project:
sh
mkdocs new my-project
cd my-project
Configure mkdocstrings in mkdocs.yml:
yaml
site_name: My Project Docs
plugins:
  - search
  - mkdocstrings
nav:
  - Home: index.md
  - Modules: 
    - module1: api/module1.md
    - module2: api/module2.md
Create Markdown Files for Each Module:
For example, create api/module1.md:
markdown
# Module 1
::: module1
Run MkDocs to Serve or Build the Documentation:
To serve locally:
sh
mkdocs serve
To build the static site:
sh
mkdocs build

**Reply by pawamoy** [reply]
Hey again @KINGOFKINGOFKEY, if you have some trouble setting up and using mkdocstrings, please open a new Q&A discussion :slightly_smiling_face:
